raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -144,6 +204,119 @@ public void testInitializeAsCandidateAndBecomeLeader() throws Exception {
             record.key(), record.value());
     }
 
+    @Test
+    public void testHandleBeginQuorumRequest() throws Exception {
+        int otherNodeId = 1;
+        int votedCandidateEpoch = 2;
+        Set<Integer> voters = Utils.mkSet(localId, otherNodeId);
+
+        MockQuorumStateStore electionStore = new MockQuorumStateStore() {
+            @Override
+            public ElectionState readElectionState() {
+                return ElectionState.withVotedCandidate(votedCandidateEpoch, otherNodeId);
+            }
+        };
+        KafkaRaftClient client = buildClient(voters, electionStore);
+        assertTrue(client.quorumState().isFollower());
+        assertTrue(client.quorumState().followerStateOrThrow().hasVoted());
+
+        BeginQuorumEpochRequestData beginQuorumEpochRequest = new BeginQuorumEpochRequestData()
+                                          .setLeaderId(otherNodeId)
+                                          .setLeaderEpoch(votedCandidateEpoch);
+        channel.mockReceive(new RaftRequest.Inbound(channel.newCorrelationId(), beginQuorumEpochRequest, time.milliseconds()));
+
+        client.poll();
+        assertTrue(client.quorumState().isFollower());
+        assertTrue(client.quorumState().followerStateOrThrow().hasLeader());
+        assertEquals(client.quorumState().followerStateOrThrow().leaderId(), otherNodeId);
+
+        List<RaftMessage> unsent = channel.drainSendQueue();
+        assertEquals(1, unsent.size());
+
+        assertTrue(unsent.get(0) instanceof RaftResponse.Outbound);
+        assertTrue(unsent.get(0).data() instanceof BeginQuorumEpochResponseData);
```

Comment:
Can we use `assertBeginQuorumEpochRequest`?

Reply:
It's actually expecting BeginQuorumResponse

Reply:
Let me add a helper here.